Windows documentation overhaul (#33582)
kubernetes/website: content/en/docs/concepts/windows/intro.md

for more details.

See [Windows OS version compatibility](#windows-os-version-support) for details on
the Windows versions that Kubernetes is tested against.

From an API and kubectl perspective, Windows containers behave in much the same
way as Linux-based containers. However, there are some notable differences in key
functionality which are outlined in this section.

### Comparison with Linux {#compatibility-linux-similarities}

Key Kubernetes elements work the same way in Windows as they do in Linux. This
section refers to several key workload enablers and how they map to Windows.

* [Pods](/docs/concepts/workloads/pods/)

  A Pod is the basic building block of Kubernetes–the smallest and simplest unit in
  the Kubernetes object model that you create or deploy. You may not deploy Windows and
  Linux containers in the same Pod. All containers in a Pod are scheduled onto a single
  Node where each Node represents a specific platform and architecture. The following
  Pod capabilities, properties and events are supported with Windows containers:

  * Single or multiple containers per Pod with process isolation and volume sharing
  * Pod `status` fields

>   * Readiness and Liveness probes

Author: It looks like we only run e2e tests for liveness and readiness probes for windows (which are marked as NodeConfromance tests)

Let me run some of the e2e test cases manually to test this out.

https://github.com/kubernetes/kubernetes/blob/12990dec4057847bf6e7e8037c27baa16e58648b/test/e2e/common/node/container_probe.go

  * postStart & preStop container lifecycle events
  * ConfigMap, Secrets: as environment variables or volumes
  * `emptyDir` volumes
  * Named pipe host mounts

>   * Resource limits

Author: The kubelet doesn't do anything with requests - only limits. The scheduler may still take requests into consideration when assigning pods to nodes.

* `kubectl exec`
* Pod and container metrics
* {{< glossary_tooltip text="Horizontal pod autoscaling" term_id="horizontal-pod-autoscaler" >}}
* {{< glossary_tooltip text="Resource quotas" term_id="resource-quota" >}}
* Scheduler preemption

### Command line options for the kubelet {#kubelet-compatibility}

The behavior of some kubelet command line options behave differently on Windows, as described below:

* The `--windows-priorityclass` lets you set the scheduling priority of the kubelet process (see [CPU resource management](/docs/concepts/configuration/windows-resource-management/#resource-management-cpu))
* The `--kubelet-reserve`, `--system-reserve` , and `--eviction-hard` flags update [NodeAllocatable](/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable)
* Eviction by using `--enforce-node-allocable` is not implemented
* Eviction by using `--eviction-hard` and `--eviction-soft` are not implemented
* A kubelet running on a Windows node does not have memory
  restrictions. `--kubelet-reserve` and `--system-reserve` do not set limits on
  kubelet or processes running on the host. This means kubelet or a process on the host

>   could cause memory resource starvation outside the node-allocatable and scheduler.

Reviewer: “This means kubelet or a process on the host could cause memory resource starvation outside the node-allocatable and scheduler.”

This sentence is difficult to parse.

* The `MemoryPressure` Condition is not implemented
* The kubelet does not take OOM eviction actions

### API compatibility {#api}

> There are no differences in how most of the Kubernetes APIs work for Windows. The
> subtleties around what's different come down to differences in the OS and container
> runtime. In certain situations, some properties on workload resources were designed
> under the assumption that they would be implemented on Linux, and fail to run on Windows.

Reviewer: This is a little bit verbose? How about:

```suggestion
There are subtle differences in the way the Kubernetes APIs work for Windows due to the OS
and container runtime. Some workload properties were designed for Linux, and fail to run on Windows.
```
